<launch>

    <!-- launch turtlesim simulator node -->
    <!-- spawns the leader turtle and calls it turtle1 -->

    <node pkg="turtlesim" type="turtlesim_node" name="sim"/>

    <!-- launch teleop node -->
    <!-- How does teleop get assigned to turtle1? -->
    <node pkg="turtlesim" type="turtle_teleop_key" name="teleop" output="screen"/>

    <!-- launch a tf broadcaster for the leader turtle -->
    <node pkg="mhered" type="tf_broadcaster.py" name="turtle1_tf_broadcaster"
        respawn="false" output="screen">
        <param name="turtle" type="string" value="turtle1" />
    </node>

    <!-- launch a tf broadcaster for the follower turtle -->
    <node pkg="mhered" type="tf_broadcaster.py" name="turtle_follower_tf_broadcaster"
        respawn="false" output="screen">
        <param name="turtle" type="string" value="turtle_follower" />
    </node>

    <!-- launch listener node -->
    <node pkg="mhered" type="tf_listener.py" name="listener"/>

</launch>
